Source/WebKit/GPUProcess/graphics/RemoteDisplayListRecorder.cpp

```diff
@@ -299,8 +299,8 @@ void RemoteDisplayListRecorder::drawFilteredImageBuffer(std::optional<RenderingR
 
     cachedSVGFilter->mergeEffects(svgFilter->effects());
 
-    auto& results = cachedSVGFilter->ensureResults([&]() {
-        auto allocator = makeUnique<ImageBufferShareableAllocator>(m_sharedResourceCache->resourceOwner());
+    auto& results = cachedSVGFilter->ensureResults([protectedThis = Ref { *this }]() {
```

Comment:
I believe we can mark the parameter of `ensureResults()` as `NOESCAPE` instead. Since the lambda runs synchronously, we shouldn't need protection here.